src/App.tsx

```diff
@@ -47,13 +47,50 @@ const useAppStyles = makeStyles(theme => {
 const App = (): JSX.Element => {
   const classes = useAppStyles(theme);
 
-  // This Hook is used to keep track of whether to render the table or CommitmentDetails
-  // const [displayDetails, setDisplayDetails] = React.useState(false);
   // This Hook is used to keep track of which CommitmentDetails should be displayed
   const [displayId, setDisplayId] = React.useState(
     window.location.hash.slice(1) || ''
   );
 
+  // This is intended to be used for more generalized purposes
+  const [history, setHistory] = React.useState([] as string[]); // May not need to be a hook
```

Comment:
I don't think you need to track the browser history.  The web browser tracks its own history.  For example, if you click the back button, the hash changes to whatever value it was before (even though the page doesn't update).  You aren't tracking that, all you need to do is read that value.